### PR TITLE
[NIC-9] Add JWT/JWKS supporting

### PR DESCRIPTION
**Pull request by V7074**

### CODE REVIEW

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwksController.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping(PUBLIC_API_V1_AUTH_URL_PATH)
+class JwksController(
+    private val jwksService: JwksService,
+) {
+
+    @GetMapping(path = [GET_JWKS_URL_PATH], produces = [MediaType.APPLICATION_JSON_VALUE])
+    fun getJwks(): Map<String, Any> {
```
Comment (AksyonovaKaterina) [inline backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwksController.kt]: =?

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwtController.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+@RestController
+@RequestMapping(PUBLIC_API_V1_AUTH_URL_PATH)
+class JwtController(
+    private val jwtService: JwtService,
+) {
+
+    @PostMapping(
+        path = [UPDATE_TOKENS_BY_REFRESH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
```
Comment (AksyonovaKaterina) [inline backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwtController.kt]: static import?

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/entity/UserEntity.kt [hunk 3]
```diff
@@ -16,4 +16,4 @@ data class UserEntity (
 
     @Field("password_hash")
     var passwordHash: String
```
Comment (AksyonovaKaterina) [inline backend/auth-service/src/main/kotlin/com/nosql/authservice/entity/UserEntity.kt]: ,

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/exception/ApplicationException.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.nosql.authservice.exception
+
+import com.nosql.authservice.common.exception.enumerator.ApplicationExceptionCodeEnum
+import org.springframework.http.HttpStatus
+
+open class ApplicationException(
```
Comment (AksyonovaKaterina) [inline backend/auth-service/src/main/kotlin/com/nosql/authservice/exception/ApplicationException.kt]: use from package com.nosql.authservice.common.exception

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwksController.kt (hunk 1 above)
Comment (V7074) [reply backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwksController.kt]: fixed

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwtController.kt (hunk 2 above)
Comment (V7074) [reply backend/auth-service/src/main/kotlin/com/nosql/authservice/controller/JwtController.kt]: fixed

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/entity/UserEntity.kt (hunk 3 above)
Comment (V7074) [reply backend/auth-service/src/main/kotlin/com/nosql/authservice/entity/UserEntity.kt]: fixed

File: backend/auth-service/src/main/kotlin/com/nosql/authservice/exception/ApplicationException.kt (hunk 4 above)
Comment (V7074) [reply backend/auth-service/src/main/kotlin/com/nosql/authservice/exception/ApplicationException.kt]: fixed

**Review by AksyonovaKaterina — APPROVED** [review]

**Pull request merged**